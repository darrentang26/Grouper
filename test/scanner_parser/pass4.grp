let int * int -> int add = function (n m) n + m
and int -> int inverse = function (n) -1 * n 
and int identity = 0
and int * int -> bool equals = function (n m) n == m  
and int group num_group = group {add identity inverse equals} 
and int mult_id = 1
and int * int -> int mult = function (n m) n * m
and int ring num_ring = ring {num_group mult mult_id}