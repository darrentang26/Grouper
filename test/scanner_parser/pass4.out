let (Int * Int) -> Int add = (n, m) -> n + m in
let Int -> Int inverse = (n) -> 1 * n in
let (Int * Int) -> Bool equals = (n, m) -> n == m in
let Int group num_group = Int add 0 inverse equals in
let (Int * Int) -> Int mult = (n, m) -> n * m in
let Int ring num_ring = Int add 0 inverse equals 1 mult in
num_ring